### PR TITLE
Use Environment Variable to Set Dark Sky Api Key

### PR DESCRIPTION
**Pull request by nadyakuzmina**
Also added `export DARK_SKY_API_KEY='<dark-sky-api-key>'` to my `.bashrc` file.

### CODE REVIEW

File: HelloWorldService/src/main/java/com/gpnk/weather/WeatherModule.java [hunk 1]
```diff
@@ -16,14 +19,18 @@ protected void config() {
...
     @Provides
     @Named("darkSkySecretKey")
     public String getDarkSkySecretKey() {
-        // TODO: get this from config
-        // This is an old api key for an account I signed up for.  It is no longer valid.
-        // Get your own key here: https://darksky.net/dev
-        return "1b06b5c498df5d71efac20db9ea91fd6";
+
+        Config config = ConfigUtil.load();
+        Config hwConfig = config.getConfig("weather");
```
Comment (mrgreg) [inline HelloWorldService/src/main/java/com/gpnk/weather/WeatherModule.java]: What does the hw stand for in hwConfig?

File: HelloWorldService/src/main/resources/application.conf [hunk 2]
```diff
@@ -50,6 +50,11 @@ helloworld {
   weatherTemplate: "Hello %s, the weather is %s!"
 }
 
+weather {
+  darkSkyApiKey: ""
+  darkSkyApiKey: ${?DARK_SKY_API_KEY}
+}
```
Comment (mrgreg) [inline HelloWorldService/src/main/resources/application.conf:56]: Do we need line 54?  What happens if it's missing?

**Review by mrgreg — APPROVED** [review]
two questions, but nothing major

File: HelloWorldService/src/main/java/com/gpnk/weather/WeatherModule.java (hunk 1 above)
Comment (nadyakuzmina) [reply HelloWorldService/src/main/java/com/gpnk/weather/WeatherModule.java]: Good point! That got left from a copy-paste. Renamed this variable to `weatherConfig`.

File: HelloWorldService/src/main/resources/application.conf (hunk 2 above)
Comment (nadyakuzmina) [reply HelloWorldService/src/main/resources/application.conf:56]: If the env var is not set and line 54 is missing, the application gets an Exception on startup saying that darkSkyApiKey is not set.

Comment (nadyakuzmina) [reply HelloWorldService/src/main/resources/application.conf:56]: I vote we keep line 54.

**Pull request merged**